docs-md/tutorials/embedded-connect.md

```diff
@@ -0,0 +1,408 @@
+`docker-compose` content into a file named `docker-compose.yml`. You will
+create and add a number of other files to this directory during this tutorial.
+
+```yaml
+---
+version: '2'
+
+services:
+  zookeeper:
+    image: confluentinc/cp-zookeeper:5.3.1
+    hostname: zookeeper
+    container_name: zookeeper
+    ports:
+      - "2181:2181"
+    environment:
+      ZOOKEEPER_CLIENT_PORT: 2181
+      ZOOKEEPER_TICK_TIME: 2000
+
+  broker:
+    image: confluentinc/cp-enterprise-kafka:5.3.1
+    hostname: broker
+    container_name: broker
+    depends_on:
+      - zookeeper
+    ports:
+      - "29092:29092"
+    environment:
+      KAFKA_BROKER_ID: 1
+      KAFKA_ZOOKEEPER_CONNECT: 'zookeeper:2181'
+      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: PLAINTEXT:PLAINTEXT,PLAINTEXT_HOST:PLAINTEXT
+      KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://broker:9092,PLAINTEXT_HOST://localhost:29092
+      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
+      KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS: 0
+
+  ksqldb-server:
+    image: confluentinc/ksqldb-server:0.6.0
+    hostname: ksqldb-server
+    container_name: ksqldb-server
+    depends_on:
+      - broker
+    ports:
+      - "8088:8088"
+    environment:
+      KSQL_LISTENERS: http://0.0.0.0:8088
+      KSQL_BOOTSTRAP_SERVERS: broker:9092
+      KSQL_KSQL_LOGGING_PROCESSING_STREAM_AUTO_CREATE: "true"
+      KSQL_KSQL_LOGGING_PROCESSING_TOPIC_AUTO_CREATE: "true"
+      KSQL_KSQL_CONNECT_WORKER_CONFIG: "/connect/connect.properties"
+    volumes:
+      - ./confluentinc-kafka-connect-jdbc-5.3.1:/usr/share/kafka/plugins/jdbc
+      - ./connect.properties:/connect/connect.properties
+
+  ksqldb-cli:
+    image: confluentinc/ksqldb-cli:0.6.0
+    container_name: ksqldb-cli
+    depends_on:
+      - broker
+      - ksqldb-server
+    entrypoint: /bin/sh
+    tty: true
+  
+  postgres:
+    image: postgres:12
+    hostname: postgres
+    container_name: postgres
+    ports:
+      - "5432:5432"
+```
+
+2. Get the JDBC connector
+-------------------------
+
+[Download the JDBC connector](https://www.confluent.io/hub/confluentinc/kafka-connect-jdbc)
+to your local working directory. Next, unzip the downloaded archive:
+
+```bash
+unzip confluentinc-kafka-connect-jdbc-5.3.1.zip
+```
+
+3. Configure Connect
+--------------------
+
+In order to tell ksqlDB to run Connect in embedded mode, we must point ksqlDB
+to a separate Connect configuration file. In our docker-compose file, this is
+done via the `KSQL_KSQL_CONNECT_WORKER_CONFIG` environment variable. From
+within your local working directory, run this command to generate the Connect
+configuration file:
+
+```bash
+cat << EOF > ./connect.properties
+bootstrap.servers=broker:9092
+plugin.path=/usr/share/kafka/plugins
+group.id=ksql-connect-cluster
+key.converter=org.apache.kafka.connect.json.JsonConverter
+value.converter=org.apache.kafka.connect.json.JsonConverter
+value.converter.schemas.enable=false
+internal.key.converter.schemas.enable=false
```

Comment:
Internal converters are not needed, they were deprecated in Apache Kafka 2.0 / KIP -174